context.go

```diff
@@ -258,6 +274,43 @@ func getAvailableContexts() ([]os.FileInfo, error) {
 	return files, nil
 }
 
+func updateContextFile(file map[string]string, key string, value string) (map[string]string, error) {
+	contextValues := map[string]string{}
+	var validContextKey bool
+
+	for k, v := range file {
```

Comment:
I feel this function is overcomplicated. As far as I can see this function updates the context with a new key-value pair.
So, according to the code `contextValue` var is literally a copy of `file` var except 2 checks:
 - value by key already exists
 - key is Fn API URL.

all you need to do is:
 - take `file`
 - see if value exists `value, ok := file[key]`
 - if yes - return an error
 - if no - check if key is Fn API URL, if so - validate
 - set kev-value

basically, you don't have to loop through the map here every time you need to update the context

Instead of creating multiple instances of `map[string]string` you better be declaring a type:
```go
type ContextMap map[string]string
```
and declare few methods for it, like:
```
func (ctxMap *ContextMap) Set(key, value string) error
func (ctxMap *ContextMap) Get(key string) string
```
This will give you better readibility and better UX while using those during CLI development if needed.